link/link_test.go

```diff
@@ -163,29 +163,34 @@ func testLink(t *testing.T, link Link, opts testLinkOptions) {
 		}
 	}
 
-	t.Run("update", func(t *testing.T) {
-		err := link.Update(opts.prog)
-		if err == ErrNotSupported {
-			t.Fatal("Update returns unwrapped ErrNotSupported", link)
-		}
-		if errors.Is(err, ErrNotSupported) {
-			return
-		}
-		if err != nil {
-			t.Fatal("Update returns an error:", err)
-		}
+	switch link.(type) {
+	case *FreplaceLink:
+		// Tracing links can't be updated
```

Comment:
`Update` should return `ErrNotSupported` in that case. Then this change isn't needed.